utils/firestore.js

```diff
@@ -1410,137 +1410,72 @@ const getSpecimenCollections = async (token, siteCode) => {
-            }
-        }
+
+    if (startDate !== '0') {
+        query = query.where('656548982', '>=', startDate);
     }
-    else{
-        if(endDate !== "0"){
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', "<=", endDate).where('656548982' ,">=", startDate).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', "<=", endDate).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-        }
-        else{
-            if(startDate !== "0"){
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).where('656548982', ">=", startDate).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-            else{
-                snapshot =  await db.collection('boxes').where('789843387', '==', siteCode).where('145971562','==',353358909).orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage*elementsPerPage).get();
-            }
-        }
+
+    if (source !== 'bptlShippingReport') {
+        query = query.where('789843387', '==', siteCode);
     }
-    let result = snapshot.docs.map(document => document.data());
-    return result;
-    /*if(snapshot.size !== 0){
-        
-        let arrSnaps = snapshot.docs;
-        let toStart = currPage * elementsPerPage;
-        let toReturnArr = snapshot.splice(toStart, toStart+25 > arrSnaps.length? arrSnaps.length : tStart + 25);
-        let toReturn = [toReturnArr, Math.ceil(arrSnaps.length)]
-        return toReturn;
+    return query
+}
+
+const getBoxesPagination = async (siteCode, body) => {
+    const currPage = body.pageNumber;
+    const orderByField = body.orderBy;
+    const elementsPerPage = body.elementsPerPage;
+    const filters = body.filters ?? {};
+    const source = body.source ?? '';
+
+    const startDate = filters.startDate ?? '0';
+    const trackingId = filters.trackingId ?? '';
+    const endDate = filters.endDate ?? '0';
+    try {
+        let query = db.collection('boxes').where('145971562', '==', 353358909);
+
+        query = buildQueryWithFilters(query, trackingId, endDate, startDate, source, siteCode)
+
+        query = query.orderBy(orderByField, 'desc').limit(elementsPerPage).offset(currPage * elementsPerPage);
+
+        const snapshot = await query.get();
+        const result = snapshot.docs.map(document => document.data());
+
+        return result;
+    } catch (error) {
+        console.error(error);
+        return new Error(error)
```

Comment:
I suggest returning `[]` (same data type as normal return), because this returned error is not handled by calling function.